src/parser/postgresparser.cpp

```diff
@@ -894,26 +894,44 @@ parser::ColumnDefinition *PostgresParser::ColumnDefTransform(ColumnDef *root) {
       else if (constraint->contype == CONSTR_UNIQUE)
         result->unique = true;
       else if (constraint->contype == CONSTR_FOREIGN) {
-        result->table_info_.reset(new TableInfo());
         // Transform foreign key attributes
         // Reference table
-        result->table_info_->table_name = constraint->pktable->relname;
-        // Reference column
+
+        // additionally, add a speci·al ColumnDefinition to the list!
+        auto col = new ColumnDefinition(ColumnDefinition::DataType::FOREIGN);
+
+        col->foreign_key_source.emplace_back(root->colname);
         if (constraint->pk_attrs != nullptr)
-          for (auto attr_cell = constraint->pk_attrs->head;
-               attr_cell != nullptr; attr_cell = attr_cell->next) {
-            value *attr_val =
-                reinterpret_cast<value *>(attr_cell->data.ptr_value);
-            result->foreign_key_sink.push_back(std::string(attr_val->val.str));
-          }
+        {
+          auto attr_cell = constraint->pk_attrs->head;
+          value* attr_val =
+                reinterpret_cast<value*>(attr_cell->data.ptr_value);
+          col->foreign_key_sink.emplace_back(attr_val->val.str);
+
+          LOG_ERROR("PK attr: %s\n", attr_val->val.str);
+        }
+        else
+        {
+          // TODO: Default to the primary key of the sink table
+
+        }
+
+        // Update Reference Table
+        col->fk_sink_table_name = constraint->pktable->relname;
         // Action type
-        result->foreign_key_delete_action =
+        col->foreign_key_delete_action =
             CharToActionType(constraint->fk_del_action);
-        result->foreign_key_update_action =
+        col->foreign_key_update_action =
             CharToActionType(constraint->fk_upd_action);
         // Match type
-        result->foreign_key_match_type =
-            CharToMatchType(constraint->fk_matchtype);
+        col->foreign_key_match_type = CharToMatchType(constraint->fk_matchtype);
+
+        LOG_ERROR("PK table name: %s.\n", col->fk_sink_table_name.c_str());
```

Comment:
On an initial read.. LOG_ERROR statements report information, but what is the error they are trying to report? Similarly for line 911.